src/amo/reducers/collections.js

```diff
@@ -0,0 +1,160 @@
+/* @flow */
+import { createInternalAddon } from 'core/reducers/addons';
+import type { AddonType, ExternalAddonType } from 'core/types/addons';
+
+export const FETCH_COLLECTION = 'FETCH_COLLECTION';
+export const LOAD_COLLECTION = 'LOAD_COLLECTION';
+
+export type CollectionType = {
+  addons: Array<AddonType>,
+  authorName: string,
+  description: string | null,
+  id: number,
+  lastUpdatedDate: string,
+  name: string,
+  numberOfAddons: number,
+};
+
+export type CollectionsState = {|
+  current: CollectionType | null,
+  loading: boolean,
+|};
+
+export const initialState: CollectionsState = {
+  current: null,
+  loading: false,
+};
+
+type FetchCollectionParams = {|
+  errorHandlerId: string,
+  page?: number,
+  slug: string,
+  user: number | string,
+|};
+
+export const fetchCollection = ({
+  errorHandlerId,
+  page,
+  slug,
+  user,
+}: FetchCollectionParams = {}) => {
```

Comment:
This should include a type for the returned action object.

Reply:
Ah right, I guess I was lazy and omitted all the Action types because there were a lot of types already.